transport/tcp/serve_tcp_ipmap.go

```diff
@@ -0,0 +1,113 @@
+package tcp
+
+import (
+	"net"
+	"sort"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/zrepl/zrepl/transport"
+)
+
+type ipMapEntry struct {
+	ip     net.IP
+	ident  string
+	subnet *net.IPNet
+	zone   string
+}
+
+type ipMap struct {
+	entries []ipMapEntry
+}
+
+func ipMapFromConfig(clients map[string]string) (*ipMap, error) {
+	entries := make([]ipMapEntry, 0, len(clients))
+	for clientInput, clientIdent := range clients {
+		clientIP, clientCIDR, zone := handleIPV6(clientInput)
+
+		if clientCIDR != nil {
+			if len(clientIdent) == 1 && clientIdent == "*" {
+				clientIdent = "willbeclientip"
```

Comment:
`"willbeclientip"` would be a legitimate user-specified client identity
=> you cannot use that as a sentinel value